### PR TITLE
explain how to run SpotBugs plugin on SonarQube

### PR DESCRIPTION
**Pull request by KengoTODA**
Recently I released SonarQube plugin that depends on sonar-findbugs. And I found that we need to care several points to make SonarQube plugin works with SpotBugs plugin, so I made this document.

You can see [staging page at here](https://spotbugs-in-kengo-toda.readthedocs.io/en/documentation/use-plugin-on-sonarqube.html).

### CODE REVIEW

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 1.3\n"
+
+#: ../../use-plugin-on-sonarqube.rst:2
+msgid "Use SpotBugs Plugin on SonarQube"
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:4
+msgid ""
+"`SpotBugs SonarQube Plugin <https://github.com/spotbugs/sonar-findbugs>`_"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: Should be "The SpotBugs SonarQube Plugin..."

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+" uses major SpotBugs plugin such as `fb-contrib <http://fb-"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: Should be "plugins" plural

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+"contrib.sourceforge.net/>`_ and `Find Security Bugs "
+"<http://h3xstream.github.io/find-sec-bugs/>`_. However, if you want to "
+"use other SpotBugs plugin, you need to build own SonarQube plugin. For "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "use another"

"build your own"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+"detailed requirements on SonarQube plugin, see `SonarQube official "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "on SonarQube plugins"

"`the SonarQube"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+"guideline <https://docs.sonarqube.org/display/DEV/Developing+a+Plugin>`_."
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "guidelines"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:7
+msgid "Create Maven Project"
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:9
+msgid ""
+"Follow interaction in `SonarQube official guideline "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "Follow the interaction"

"`the SonarQube official guidelines"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+"<https://docs.sonarqube.org/display/DEV/Build+Plugin#BuildPlugin-"
+"CreateaMavenProject>`_. It is recommended to use sub-module, to manage "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "use sub-modules"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+"both of SpotBugs plugin and SonarQube plugin in one project. You can "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "both the SpotBugs plugin and the SonarQube plugin"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+"refer `this module <https://github.com/KengoTODA/guava-helper-for-"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "refer to `this module"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+"java-8/tree/master/sonarqube-plugin>`_ as example."
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "as an example"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:12
+msgid ""
+"You also need to configure ``sonar-packaging-maven-plugin``, to make your"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "to configure the"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+" plugin depends on `SpotBugs SonarQube Plugin "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "depend on the"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+"<https://github.com/spotbugs/sonar-findbugs>`_. For instance, if you're "
+"using SonarQube 6.7 LTS, your plugin requires SpotBugs SonarQube Plugin "
+"version 3.7, so configuration should be like below:"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "so your configuration"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+#: ../../use-plugin-on-sonarqube.rst:23
+msgid "Generate rules.xml"
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:25
+msgid ""
+"SonarQube doesn't understand the Bug Pattern metadata provided for "
+"SpotBugs, so we need to convert ``findbugs.xml`` and ``messages.xml`` to "
+"SonarQube format named ``rules.xml``."
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "the SonarQube format"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:27
+msgid ""
+"If your SpotBugs plugin isn't complex, you can simply introduce "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "introduce the"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+"`SonarQube rule xml generator Maven Plugin <https://github.com/KengoTODA"
+"/sonarqube-rule-xml-generator>`_ to generate ``rules.xml``. Follow `the "
+"interaction described in its README <https://github.com/KengoTODA"
+"/sonarqube-rule-xml-generator#how-to-use>`_."
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:30
+msgid "Update RulesDefinision.java"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: Shouldn't this be `RulesDefinition.java`?

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:32
+msgid ""
+"Your ``SonarQubeRulesDefinition.java`` should load generated "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "load the generated"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+"``rules.xml`` to FindBugs repository."
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "to your FindBugs repository"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:34
+msgid ""
+"When you create a ``NewRepository`` instance, use "
+"``FindbugsRulesDefinition.REPOSITORY_KEY`` as repository key, and do "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "as the repository key"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+"_not_ rename it by calling ``NewRepository#setName(String)``. It is "
+"necessary to fulfill the requirement from `SonarQube API "
+"<https://github.com/SonarSource/sonarqube/blob/6.7.4/sonar-plugin-"
+"api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java#L393-L395>`_."
+" Here is example:"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "Here is an example"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:53
+msgid "Update Plugin.java"
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:55
+msgid ""
+"``Plugin.java`` should be simple implementation that just loads your "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "should be a simple implementation"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+"``RulesDefinition`` class. Here is example:"
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "Here is an example"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:65
+msgid "Deploy onto SonarQube"
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:67
+msgid ""
+"``mvn package`` will generate ``.jar`` file that works as SonarQube "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "generate a ``.jar`` file"

"works as a SonarQube"

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+"plugin. Follow `SonarQube official guideline "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "Follow `the SonarQube official guidelines"

File: docs/use-plugin-on-sonarqube.rst [hunk 25]
```diff
@@ -0,0 +1,69 @@
+Use SpotBugs Plugin on SonarQube
+================================
+
+`SpotBugs SonarQube Plugin <https://github.com/spotbugs/sonar-findbugs>`_ uses major SpotBugs plugin such as `fb-contrib <http://fb-contrib.sourceforge.net/>`_ and `Find Security Bugs <http://h3xstream.github.io/find-sec-bugs/>`_. However, if you want to use other SpotBugs plugin, you need to build own SonarQube plugin. For detailed requirements on SonarQube plugin, see `SonarQube official guideline <https://docs.sonarqube.org/display/DEV/Developing+a+Plugin>`_.
```
Comment (ThrawnCA) [inline docs/use-plugin-on-sonarqube.rst]: See comments on the `.po` file

**Comment by ThrawnCA** [comment]
Thanks for writing this :). Just added some grammatical feedback.

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po (hunk 18 above)
Comment (KengoTODA) [reply docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: It's not 'your' but the repository generated by [sonar-findbugs](https://github.com/spotbugs/sonar-findbugs).
Maybe we can say 'existing' repository?

Comment (ThrawnCA) [reply docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: Just "the FindBugs repository" should be fine then.

File: docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po [hunk 26]
```diff
@@ -0,0 +1,127 @@
...
+"CreateaMavenProject>`_. It is recommended to use sub-modules, to manage "
+"both the SpotBugs plugin and the SonarQube plugin in one project. You can"
+" refer to `this module <https://github.com/KengoTODA/guava-helper-for-"
+"java-8/tree/master/sonarqube-plugin>`_ as an example."
+msgstr ""
+
+#: ../../use-plugin-on-sonarqube.rst:12
+msgid ""
+"You also need to configure the ``sonar-packaging-maven-plugin``, to make "
+"your plugin depends on `the SpotBugs SonarQube Plugin "
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/use-plugin-on-sonarqube.po]: "your plugin depend on"

**Comment by KengoTODA** [comment]
Updated, thank you!

**Pull request merged**